### PR TITLE
feat: Exotel integration & Call popup enhancements

### PR DESCRIPTION
**Pull request by surajshetty3416**
- Updated Call Log doctype
  - New fields (Start Time, End Time, Summary, Links, Type)
  - Removed fields (Contact, Lead) (these will be maintained in Links child table)
  - Change type of Duration from `Int` to `Duration`

Screenshot 2020-06-23 at 9 09 58 AM

- Ability to trigger a call from the app.

	call_popup_trigger

- Call Log can be linked with multiple documents. Linked Call Log will be shown in document's timeline.
Screenshot 2020-06-23 at 8 50 52 AM

- A site-specific unique link to set up call popup in Exotel settings for security.
Screenshot 2020-06-23 at 9 14 38 AM

- Call Popup update - Now shows more information
	For unknown caller:
	Screenshot 2020-06-23 at 11 11 11 AM
	
	For known caller:
	Screenshot 2020-06-23 at 11 03 16 AM
	Screenshot 2020-06-23 at 11 02 49 AM
	Screenshot 2020-06-23 at 11 02 41 AM

Requires: https://github.com/frappe/frappe/pull/10677

BREAKING CHANGE

### CODE REVIEW

File: erpnext/patches/v13_0/move_call_reference_to_links.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import frappe
+
+# move employee/lead/Contact to links
+def execute():
+	frappe.reload_doctype('Call Log')
+
+	frappe.db.sql("UPDATE `tabCall Log` SET `type`='Incoming' where `type` is NULL")
+	frappe.db.sql("UPDATE `tabCall Log` SET `status`='No Answer' where `status` is 'Missed'")
```
Comment (Thunderbottom) [inline erpnext/patches/v13_0/move_call_reference_to_links.py]:
```suggestion
	frappe.db.sql("UPDATE `tabCall Log` SET `status`='No Answer' where `status`='Missed'")
```

**Comment by surajshetty3416** [comment]
Will re-open once the doc is ready.

**Comment by harshit-30** [comment]
@surajshetty3416 can we also fetch the User's or Employee's number to showcase who actually answered? Right now it reflects only numbers.

**Comment by surajshetty3416** [comment]
Parts of this PR have been moved to the telephony module via https://github.com/frappe/erpnext/pull/24032 & https://github.com/frappe/erpnext/pull/24485.

**Comment by ghostersk** [comment]
We have already PBX IP Phone system in place in the UK, I wonder if is possible to use it with ERPNext? I know about Exotel but that is another service we would have to pay no? Can we use our existing phone line to be recorded in ERPNext? Thank you

**Comment by D-codE-Hub** [comment]
Can I get Call popup in exotel integration

**Pull request closed without merging**